ethcore/blockchain/src/generator.rs

```diff
@@ -166,11 +191,19 @@ impl BlockBuilder {
 			let mut block = Block::default();
 			let metadata = get_metadata();
 			let block_number = parent_number + 1;
+			let transactions = metadata.transactions;
+			let transactions_root = ordered_trie_root(transactions.iter().map(|tx| {
```

Comment:
You sure you need to loop over all txs and encode them like this, and why as a stream and not `rlp::encode`? Honest question, I really don't know!

Reply:
yeh, `.map(rlp::encode)` produces the same result and is shorter